topics/reference_guide/custom_language_support/websymbols_context.md

```diff
@@ -0,0 +1,257 @@
+<!-- Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license. -->
+
+# Web Symbols Context
+
+<link-summary>How to use Web Symbols context detection to manage enablement of plugin features.</link-summary>
+
+One of the important qualities of well-written plugins is the enablement of its features
+only when needed. For instance, if a user does not have a particular web framework in their project,
+HTML files should not contain that framework-specific assistance.
+
+Web Symbols framework provides various ways to detect current context
+and retrieve it through `WebSymbolsContext.get(kind, ...)` methods.
+
+For a particular `kind` of Web Symbol context, there is only one `name` detected, and it can be `null`, if
+the context `kind` is not present in the location.
+
+## Location - `PsiElement` vs `VirtualFile`
+
+The location can either be a `PsiElement` or a `VirtualFile` with a `Project`. If the location is `PsiElement`,
```

Comment:
```suggestion
The location can either be a `PsiElement`, or a `VirtualFile` with a `Project`. If the location is `PsiElement`,
```
Not sure about it. Without comma, it sounds like both PsiElement and VirtualFile are subject of "with a Project".